spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/StreamTemplate.java

```diff
@@ -83,16 +75,15 @@ public StreamDefinitionResource createStream(String name, String definition, boo
 
 	@Override
 	public void deploy(String name, Map<String, String> properties) {
-		String uri = deploymentLink.expand(name).getHref();
+		String uri = deploymentsLink.getHref() + "/{name}";
```

Comment:
yeah, it was because the relation needs to be `streams/definition` when the links are added in AdminController. After fixing this link in AdminController, updated the changes that expand the Link to get the appropriate URI template.